src/dxtbx/format/cbf_writer.py

```diff
@@ -209,6 +209,7 @@ def recursive_setup_basis_dict(key, parent_name="", panel_id=0):
                 diffrn_id,
                 False,
                 gain=[panel.get_gain() for panel in detector],
+                flux=beam.get_flux(),
```

Comment:
Worth explicitly checking if `beam.get_flux()` is not None?

Reply:
That's checked in the function being called.